### ISSUE TITLE
Comment can crash Powershell Core 7 rc1

### ISSUE THREAD

**Issue by pfmoore** [PowerShell/PowerShell#11425]
# Steps to reproduce

```powershell
# iex (New-Object Net.WebClient).DownloadString("https://gist.github.com/darkoperator/6152630/raw/c67de4f7cd780ba367cccbc2593f38d18ce6df89/instposhsshdev")

```

Minimal example I can find:
```powershell

# iex (New-Object Net.WebClient).DownloadString("https://x")
```

It seems to require a HTTPS URL - any URL, even if not valid. I'm not clear why it's even executing any code, given that this is a comment.

# Expected behavior

Nothing. This is a **comment** and should not do anything.

# Actual behavior

Powershell instance terminates

# Environment data

Powershell 7 rc1 MSI downloaded today and installed to the default location.

```none
PS C:\Work\Scratch> $PSVersionTable                          
                                                             
Name                           Value                         
----                           -----                         
PSVersion                      7.0.0-rc.1                    
PSEdition                      Core                          
GitCommitId                    7.0.0-rc.1                    
OS                             Microsoft Windows 10.0.17763  
Platform                       Win32NT                       
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}         
PSRemotingProtocolVersion      2.3                           
SerializationVersion           1.1.0.1                       
WSManStackVersion              3.0                           
```

**Comment by kilasuit** [comment]
I've tried to replicate however from the steps provided I can't replicate and get the expected behaviour

**Comment by SteveL-MSFT** [comment]
This doesn't repro for me

**Comment by pfmoore** [comment]
Hmm, I've confirmed on a different machine that it doesn't crash there.

I'm not clear how a comment can cause any execution of code. Could a virus checker pick up on something like this and error somehow? If so, why would it only happen in Powershell core? It doesn't happen in Windows Powershell 5.1 :-(

**Comment by SteveL-MSFT** [comment]
@pfmoore is it possible for you to get a stack trace?  Also try disabling your virus scanner and see if it repros.

**Comment by pfmoore** [comment]
I probably won't be able to disable the virus scanner (it's a corporate PC and the virus scanner is managed globally) but I can try to get a stack trace if you can explain how I would do that - at the moment all that happens is that the powershell instance terminates with no message or anything.

I do have visual studio, so if it's possible to run powershell under a debugger using that I can do so.

**Comment by SteveL-MSFT** [comment]
@pfmoore yes, Visual Studio will work.  Just follow the steps here: https://docs.microsoft.com/en-us/visualstudio/debugger/debug-using-the-just-in-time-debugger?view=vs-2019

**Comment by pfmoore** [comment]
Looks like I mis-remembered. I don't have full Visual Studio on that PC, just the "build tools". Is there a way to get a stack trace without full Visual Studio?

**Comment by SteveL-MSFT** [comment]
You can use the free WinDbg: https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugger-download-tools

**Comment by pfmoore** [comment]
Sorry for the delay. I couldn't get the new WinDbg Preview (the store app) to register as a just-in-time debugger, but I did find SysInternals procdump, and registered that.

However, triggering the failure did not result in a dump file - whether that's because Powershell doesn't actually crash, but just terminates with no message, I'm not sure.

I did check the event viewer, and found the following message in the application log:

> UK03306 ran C:\Program Files\PowerShell\7-preview\pwsh.exe. The Trojan named AMSI-FHR!68F3442B9E26 was detected and blocked.

So it looks like it may be a false report from the virus checker. But I'm still unclear why this is only happening in Powershell Core, and not in the default Windows Powershell.
